### PR TITLE
[CEC] fix CECToggleState, screensaver looping etc. 

### PR DESCRIPTION
**Pull request by Glenn-1990**
This one superseeds https://github.com/xbmc/xbmc/pull/10492 and contains only the most important code changes. Setting that don't apply for unsuspendable devices (RPI, IMX, etc.) are not shown to avoid user confusion.

Issues that should be solved now:
http://forum.kodi.tv/showthread.php?tid=225779
https://discourse.osmc.tv/t/control-tv-on-standby-toggle-using-cec-kodi-built-in-functions/3638/2

@opdenkamp

### CODE REVIEW

**Comment by kib** [comment]
ping @opdenkamp

File: xbmc/peripherals/devices/PeripheralCecAdapter.cpp [hunk 1]
```diff
@@ -1185,12 +1185,12 @@ void CPeripheralCecAdapter::CecSourceActivated(void *cbParam, const CEC::cec_log
...
     }
-    else if (bPlayingAndDeactivated
-      && adapter->GetSettingInt("pause_or_stop_playback_on_deactivate") == LOCALISED_ID_STOP)
+    else if (activated == 0 && adapter->GetSettingInt("pause_or_stop_playback_on_deactivate") == LOCALISED_ID_STOP)
     {
       if (pSlideShow)
         pSlideShow->OnAction(CAction(ACTION_STOP));
-      else
+      else if (g_application.m_pPlayer->IsPlaying())
+        // only when playing as TMSG_MEDIA_STOP will wake the screensaver, this can make the source active again
```
Comment (Jalle19) [inline xbmc/peripherals/devices/PeripheralCecAdapter.cpp]: Perhaps a better solution would be to not wake the screen saver when stop is pressed? Most people just press any button I think, I doubt anyone would even notice if stop doesn't work.

File: system/peripherals.xml [hunk 2]
```diff
@@ -25,7 +25,6 @@
     <setting key="connected_device" type="enum" label="36019" value="36037" lvalues="36037|36038" order="12" />
     <setting key="cec_hdmi_port" type="int" value="1" min="1" max="15" label="36015" order="13" />
     <setting key="physical_address" type="string" label="36021" value="0" order="14" />
-    <setting key="port" type="string" value="" label="36022" order="15" />
```
Comment (Jalle19) [inline system/peripherals.xml]: Is the string (36022) used anywhere else? If not, drop that one as well.

File: xbmc/peripherals/devices/PeripheralCecAdapter.cpp (hunk 1 above)
Comment (Glenn-1990) [inline xbmc/peripherals/devices/PeripheralCecAdapter.cpp]: I'm not sure, I do not want break existing logic. In any case, we sending "stop" when we are not playing isn't needed.

File: system/peripherals.xml (hunk 2 above)
Comment (Glenn-1990) [inline system/peripherals.xml]: Indeed, that seems to be the case ;-)

**Comment by tudstudent** [comment]
@Glenn-1990 , would this resolve my issue which I posted here:
http://trac.kodi.tv/ticket/16842

If it is I can test this PR tomorrow/thursday. Would be very welcome if it solves the standby issue on my set.

**Comment by Glenn-1990** [comment]
@tudstudent Probably not, can you post a logfile with debugging enabled in Kodi (when sending Kodi to sleep and CEC devices stay on)?

**Comment by tudstudent** [comment]
@Glenn-1990 I have uploaded a log including CEC peripheral logging:
http://paste.ubuntu.com/23250235/
Hope you can help with this problem

File: xbmc/peripherals/devices/PeripheralCecAdapter.cpp (hunk 1 above)
Comment (opdenkamp) [inline xbmc/peripherals/devices/PeripheralCecAdapter.cpp]: I don't mind, either works for me. Not using the stop button to wake up the screensaver myself ;-)

**Comment by MartijnKaijser** [comment]
is this PR obsolete or does it need updating ?

**Comment by opdenkamp** [comment]
It'll need updating as people are still having issues. Just got an email from @olympia about this problem earlier week. Looking at CEC related issues atm.

**Comment by olympia** [comment]
Indeed personally I am also having the same issues what @tudstudent is describing in the above trac ticket backed up with logs. If I remember correctly there were different opinions in the forums whether this is a libcec or Kodi issue, but was definitely working well in Kodi 16

**Comment by Rechi** [comment]
@Glenn-1990 this needs a rebase

**Comment by Glenn-1990** [comment]
@opdenkamp You are talking about "this problem" while this PR covers more issues, are you talking about the "CECToggleState" issue? 
My CEC enabled development environment is gone and I'm limited in time. Can you pick the commits you need?

**Pull request closed without merging**